src/pages/EnablePayments/EnablePayments.tsx

```diff
@@ -0,0 +1,86 @@
+import React, {useEffect} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import Navigation from '@libs/Navigation/Navigation';
+import * as Wallet from '@userActions/Wallet';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type {BankAccountList, UserWallet} from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import AddBankAccount from './AddBankAccount/AddBankAccount';
+import FailedKYC from './FailedKYC';
+import FeesAndTerms from './FeesAndTerms/FeesAndTerms';
+import PersonalInfo from './PersonalInfo/PersonalInfo';
+import VerifyIdentity from './VerifyIdentity/VerifyIdentity';
+
+type EnablePaymentsPageOnyxProps = {
+    /** The user's wallet */
+    userWallet: OnyxEntry<UserWallet>;
+
+    /** The list of bank accounts */
+    bankAccountList: OnyxEntry<BankAccountList>;
+};
+
+type EnablePaymentsPageProps = EnablePaymentsPageOnyxProps;
+
+function EnablePaymentsPage({userWallet, bankAccountList}: EnablePaymentsPageProps) {
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+
+    useEffect(() => {
+        if (isOffline) {
+            return;
+        }
+
+        if (isEmptyObject(userWallet)) {
+            Wallet.openEnablePaymentsPage();
+        }
+    }, [isOffline, userWallet]);
+
+    if (isEmptyObject(userWallet)) {
+        return <FullScreenLoadingIndicator />;
+    }
+
+    if (userWallet?.errorCode === CONST.WALLET.ERROR.KYC) {
+        return (
+            <>
+                <HeaderWithBackButton
+                    title={translate('additionalDetailsStep.headerTitle')}
+                    onBackButtonPress={() => Navigation.goBack(ROUTES.SETTINGS_WALLET)}
+                />
+                <FailedKYC />
+            </>
+        );
+    }
+
+    const currentStep = isEmptyObject(bankAccountList) ? CONST.WALLET.STEP.ADD_BANK_ACCOUNT : userWallet?.currentStep || CONST.WALLET.STEP.ADDITIONAL_DETAILS;
```

Comment:
We need to adjust a code which should work correctly on the back from step 1. 

**Bug**:

https://github.com/Expensify/App/assets/14358475/d32a4d0b-24c1-4a1b-b19b-eebb037a3ad4